### ISSUE TITLE
Automatically rewrite editor numbers

### ISSUE THREAD

**Issue by ivxvm** [davidxn/rampart#1]
Currently, if multiple mappers use custom decorate classes with editor numbers, they can run into collisions (not sure if these collisions are a big deal, but there are startup console errors at least). Would be a nice feature if RAMPART could automatically resolve editor number conflicts by overwriting them in individual maps and corresponding decorate files (for example, by assigning to each mapper a unique range of available editor numbers and pulling from that range).

**Comment by davidxn** [comment]
Yes, this is something I've just run into as I've been pulling RAMP 2022 together! Some control over editor numbers would definitely be good. At the moment I only interpret and warn about conflicts through doomednums in MAPINFO, but it's possible to sneak past that check by specifying them in DECORATE.

Same goes for spawnnums, which can be specified in MAPINFO or in class properties

**Comment by tsjsr3** [comment]
since editor numbers can only be specified in DECORATE and not in ZSCRIPT then maybe a function could be made that scans the decorate code for editor numbers and rejects if it finds any with the message that they must be defined in mapinfo?
